pulsar-client/src/main/java/org/apache/pulsar/client/impl/TransactionMetaStoreHandler.java

```diff
@@ -60,6 +63,10 @@
     private final ConcurrentLongHashMap<OpBase<?>> pendingRequests =
         new ConcurrentLongHashMap<>(16, 1);
     private final ConcurrentLinkedQueue<RequestTime> timeoutQueue;
+    private final ConcurrentLongHashMap<OpBase<?>> waitingExecutedRequests =
```

Comment:
Can we reuse the `pendingRequests` ? we can only remove the op from the `pendingRequests` if the request succeeds or encounter a non-recoverable exception, after the connection opened, we can just flush out the `pendingRequests` again.